sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/perPartitionCircuitBreaker/GlobalPartitionEndpointManagerForPerPartitionCircuitBreaker.java

```diff
@@ -70,168 +77,192 @@ public void init() {
         }
     }
 
-    public void handleLocationExceptionForPartitionKeyRange(RxDocumentServiceRequest request, RegionalRoutingContext failedRegionalRoutingContext) {
+    public void handleLocationExceptionForPartitionKeyRange(
+        RxDocumentServiceRequest request,
+        RegionalRoutingContext failedRegionalRoutingContext,
+        boolean isCancellationException) {
 
-        checkNotNull(request, "Argument 'request' cannot be null!");
-        checkNotNull(request.requestContext, "Argument 'request.requestContext' cannot be null!");
+        try {
+            checkNotNull(request, "Argument 'request' cannot be null!");
+            checkNotNull(request.requestContext, "Argument 'request.requestContext' cannot be null!");
 
-        PartitionKeyRange resolvedPartitionKeyRangeForCircuitBreaker = request.requestContext.resolvedPartitionKeyRangeForCircuitBreaker;
-        PartitionKeyRange resolvedPartitionKeyRange = request.requestContext.resolvedPartitionKeyRange;
+            PartitionKeyRange resolvedPartitionKeyRangeForCircuitBreaker = request.requestContext.resolvedPartitionKeyRangeForCircuitBreaker;
+            PartitionKeyRange resolvedPartitionKeyRange = request.requestContext.resolvedPartitionKeyRange;
 
-        // in scenarios where partition is splitting or invalid partition then resolvedPartitionKeyRange could be set to null
-        // no reason to circuit break a partition key range which is effectively won't be used in the future
-        if (resolvedPartitionKeyRangeForCircuitBreaker != null && resolvedPartitionKeyRange == null) {
-            return;
-        }
+            // in scenarios where partition is splitting or invalid partition then resolvedPartitionKeyRange could be set to null
+            // no reason to circuit break a partition key range which is effectively won't be used in the future
+            if (resolvedPartitionKeyRangeForCircuitBreaker != null && resolvedPartitionKeyRange == null) {
+                return;
```

Comment:
shall we log these cases? with the partition key range information which is splitting? how can a partition key range be invalid? during split requests?